### DISCUSSION TITLE
Make root URL configurable from Environment

### DISCUSSION THREAD

**Discussion by junderw** [teslamate-org/teslamate#1902] [Ideas]
I was able to get it working in grafana using these settings:
https://grafana.com/docs/grafana/v8.0/administration/configuration/#root_url
https://grafana.com/docs/grafana/v8.0/administration/configuration/#serve_from_sub_path

But Teslamate is generating links based on the root path: ie. `https://mydomain/settings` instead of `https://mydomain/mysubpath/settings` (my reverse proxy is forwarding `https://mydomain/mysubpath/*` to teslamate with my subpath chopped off ie. `http://localhost:4000/*`

This is slightly related to #1893 I am guessing.

**Comment by metaben** [comment]
I reported this as #2056, because the implementation is currently ambiguous/inconsistent.

{base_url} has been implemented as {return_from_grafana_url}. But by name, {base_url} should be the user-configurable path to the TeslaMate Web App itself.

**Comment by JaxonWright** [comment]
This functionality would probably help a lot of people out, including me. I doubt that I am the only one who wants to reverse proxy Teslamate to a subpath  (`example.com/teslamate`) and not a subdomain (`teslamate.example.com`).

**Reply by JaxonWright** [reply]
@adriankumpf how difficult do you foresee something like this? There is already a setup option that I thought this was for (URLs > Web App), but it is not. I don't see a scenario where that couldn't be used for the base URL of all of the hrefs in the web app.

**Reply by adriankumpf** [reply]
This is a good idea, but I'm not up for implementing it right now. If you or someone else wants to send a PR I'd be happy to review!

**Comment by inigoml** [comment]
Same problem here. All icons and non html stuff is lost when working with teslamate app (not grafana) behind a nginx proxy server.
